elastic/beats
# Include fields from x-pack processors and autodiscover providers

## What does this PR do?

Include fields definitions of x-pack processors and autodiscover providers.

It does the following:
* Add `x-pack/libbeat/processors` and `x-pack/libbeat/autodiscover/providers` to the list of common modules in the fields generator, this effectively add these fields to the fields docs of all beats.
* Add processors and autodiscover providers as "modules" of libbeat, this adds the fields to the `fields.yml` files, but doesn't seem enough to include the fields in the binaries.

## Why is it important?

Documentation and index templates should contain these fields, they are not being included now.

## Checklist


## Author's Checklist

- [ ] Ensure that fields are included also in index templates.

## How to test this PR locally

- Check that fields for x-pack processors and autodiscover providers are included in:
  - Documentation
  - Index template (`metricbeat export template`).
